pkg/tsdbctl/create.go

```diff
@@ -152,3 +149,52 @@ func (cc *createCommandeer) validateFormat(format string) error {
 	}
 	return nil
 }
+
+func (cc *createCommandeer) rateInHours() (int, error) {
+	if len(cc.sampleRate) >= 3 {
+		last := cc.sampleRate[len(cc.sampleRate)-1:]
+		// get the number ignoring slash and time unit
+		cc.sampleRate = cc.sampleRate[0 : len(cc.sampleRate)-2]
+		i, err := strconv.Atoi(cc.sampleRate)
+		if err != nil {
+			return 0, errors.Wrap(err, `not a valid rate. Accepted pattern: [0-9]+/[dhms]. Examples: 12/m`)
+		}
+		switch last {
+		case "s":
+			return i * 60 * 60, nil
+		case "m":
+			return i * 60, nil
+		case "h":
+			return i, nil
+		case "d":
+			return i / 24, nil
+		default:
+			return 0, fmt.Errorf(`not a valid rate. Accepted pattern: [0-9]+/[dhms]. Examples: 12/m`)
+		}
+	}
+
+	return 0, fmt.Errorf(`not a valid rate. Accepted pattern: [0-9]+/[dhms]. Examples: 12/m`)
+}
+
+// This method calculates the chunk and partition interval from the given sample rate.
+func (cc *createCommandeer) getPartitionAndChunkInterval() (string, string, error) {
```

Comment:
This would probably be better as a function from rate. Doesn't need the whole commandeer thing.

Reply:
That clarity would make the comment redundant :)